otter/test/util/test_zk.py

```diff
@@ -99,40 +101,77 @@ def exists(self, path):
         else:
             return None
 
-    def Lock(self, path):
-        return ZKLock(self, path)
-
 
-class ZKLock(object):
-    """
-    Stub for :obj:`kazoo.recipe.lock.KazooLock` and :obj:`PollingLock`
-    since ``PollingLock`` provides ``KazooLock`` interface.
+class _ZKLock(object):
     """
+    Stub for :obj:`kazoo.recipe.lock.KazooLock` and :obj:`PollingLock`.
+    It provides *_eff implementations based on ``LockBehavior`` for
+    ``PollingLock``
 
-    def __init__(self, client, path):
-        self.client = client
-        self.path = path
-        self.acquired = False
-        # tuple of (blocking, timeout, return_value) to be set by test
-        self.acquire_call = ()
-        # release return value
-        self.release_call = None
+    This class is private. Get its object and control it by calling
+    ``create_fake_lock``
+    """
+    def __init__(self, behavior):
+        self._behavior = behavior
+
+    def is_acquired(self):
+        return succeed(self._behavior.acquired)
+
+    def is_acquired_eff(self):
+        return Effect(Constant(self._behavior.acquired))
+
+    def acquire_eff(self, blocking, timeout):
+        assert (self._behavior.acquired is LockBehavior.NOT_STARTED or
+                (not self._behavior.acquired))
+        assert (blocking, timeout) == self._behavior.acquire_call[:2]
+        ret = self._behavior.acquire_call[-1]
+        if isinstance(ret, Exception):
+            self._behavior.acquired = False
```

Comment:
This block is supposedly un-covered by tests, according to codecov.  I'm not sure which one would hit it.